azurerm/resource_arm_analysis_services_server.go

```diff
@@ -0,0 +1,449 @@
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmAnalysisServicesServer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAnalysisServicesServerCreate,
+		Read:   resourceArmAnalysisServicesServerRead,
+		Update: resourceArmAnalysisServicesServerUpdate,
+		Delete: resourceArmAnalysisServicesServerDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAnalysisServicesServerName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			"sku": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"D1",
+					"B1",
+					"B2",
+					"S0",
+					"S1",
+					"S2",
+					"S4",
+					"S8",
+					"S9",
+				}, false),
+			},
+
+			"admin_users": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			//"backup_blob_container_uri": {
+			//	Type:         schema.TypeString,
+			//	Optional:     true,
+			//	Sensitive:    true,
+			//	ValidateFunc: validate.URLIsHTTPOrHTTPS,
+			//},
+
+			"enable_power_bi_service": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"ipv4_firewall_rule": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"range_start": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validate.IPv4Address,
+						},
+						"range_end": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validate.IPv4Address,
+						},
+					},
+				},
+			},
+
+			"querypool_connection_mode": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validateQuerypoolConnectionMode(),
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmAnalysisServicesServerCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).analysisServicesServerClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for Azure ARM Analysis Services Server creation.")
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
```

Comment:
_minor_ Ideally we should call this `resourceGroup` for consistency

Reply:
renamed the variable in all places